content/general/events.textile

```diff
@@ -0,0 +1,374 @@
+
+h3(#encoding). Payload Encoding
+
+The encoding of payloads sent is defined when setting up a Reactor Rule in the "Reactor tab of your app":https://support.ably.io/solution/articles/3000074406. You can have the message sent in JSON format, or as a "MessagePack":https://msgpack.org payload.
+
+* "JSON":https://www.json.org (JavaScript Object Notation): An efficient data-interchange format which is fairly standard and provides simple text based encoding.
+* "MessagePack":https://msgpack.org: An efficient binary serialization format that is similar to JSON, but faster and smaller.
+
+h2(#security). WebHook security
+
+We encourage customers to use a secure HTTPS URL when configuring their Reactor Events. This will ensure that requests cannot be intercepted and all communication with your servers is secured with TLS.
+
+However, in addition, we optionally support a signature included as an HTTP header @X-Ably-Signature@ in "batched":#batching requests. The endpoint can use the chosen private API key to verify the authenticity of the WebHook data.
+
+In order to verify the signature, you need to do the following:
+
+* start with the WebHook request body. This will be a JSON string encoded with content-encoding @utf-8@;
+* identify the key based on the @keyId@ indicated in the @X-Ably-Key@ header;
+* calculate the HMAC of that request body with algorithm SHA-256 and the key being the corresponding @keyValue@ (the secret part of the key after the "@:@");
+* encode the resulting HMAC using RFC 3548 base 64;
+* compare that result with the signature value indicated in the @X-Ably-Signature@ header
+
+h3(#example-signature). WebHook HMAC SHA-256 signature verification example
+
+If you choose to sign your WebHook requests, we recommend you try the following first:
+
+# "Set up a free RequestBin HTTP endpoint test URL":https://requestbin.com/
+# "Configure a WebHook":#configure with the URL set to the RequestBin endpoint, and ensure you have chosen to "batch":#batching messages and are using a key to sign each WebHook request
+# Trigger an event using the "Dev Console":https://support.ably.io/support/solutions/articles/3000062195 in your app dashboard which will generate a WebHook. You should then confirm that the WebHook has been received in your RequestBin
+# Check that the @X-Ably-Signature@ header in your WebHook request matches the HMAC SHA-256 you create using our "Javascript HMAC SHA-256 demo":<%= JsBins.url_for('reactor/webhook-hmac-sha-256') %>
+
+h2(#examples). Examples
+
+Given the various potential combinations of @enveloped@, @batched@ and message sources, it can be good to know what to expect given certain combinations of rules.
+
+All messages sent from a Reactor Event will contain the following headers:
+
+- host := the URL of the endpoint this message has been sent to
+- content-type := the type of the payload. This will be @text/json@ or @application/x-msgpack@, depending on what Encoding you chose in your Reactor Rule
```

Comment:
- `application/json`, not `text/json`.

- that sentence is true for enveloped rules. For unenveloped, the content-type might be `application/json`, `text/plain`, or `application/octet-stream`, depending on whether the data is json, text, or binary.